library_generation/utilities.sh

```diff
@@ -18,12 +18,38 @@ __get_config_from_BUILD() {
   if_match=$5
 
   result="${default}"
-  if grep -A 15 "${rule}" "${build_file}" | grep -q "${pattern}"; then
+  if grep -A 20 "${rule}" "${build_file}" | grep -q "${pattern}"; then
     result="${if_match}"
   fi
   echo "${result}"
 }
 
+__get_iam_policy_from_BUILD() {
```

Comment:
Do we still need to search for these protos from BUILD now that they are being passed in as parameters? 
In addition, do we need to read anything else from BUILD file other than searching for common protos?

Reply:
> Do we still need to search for these protos from BUILD now that they are being passed in as parameters?

All functions that read BUILD are used by `generate_library_integration_test.sh` for testing against googleapis-gen.

> In addition, do we need to read anything else from BUILD file other than searching for common protos?

`transport`, `rest_numeric_enums` and `include_samples` are also read from BUILD.

Reply:
> All functions that read BUILD are used by generate_library_integration_test.sh for testing against googleapis-gen. They are not in test_utilities.sh because they are part of generate_sdk_libraries.sh later.

I'm not sure we need it in `generate_sdk_libraries.sh` either. We would still need to keep the bazel interface, so for a single client library, the flow would be `bazel build` -> `java_gapic_assembly_gradle_pkg` -> `generate_library.sh`, where `java_gapic_assembly_gradle_pkg` would pass all the info from bazel file to our shell script. So `generate_sdk_libraries.sh` would just be calling `bazel build` for each service in this case.

Reply:
If that's the case then we probably don't need these utility functions.

Right now it's only used in testing and can be removed without impacting `generate_library.sh`.

Reply:
> Right now it's only used in testing and can be removed without impacting generate_library.sh.

We can still keep them and move them to the test utils.

Reply:
> We can still keep them and move them to the test utils.

How about I create another PR to refactor test utilities into `test_utilities.sh`. I think there are other functions can move into `test_utilities.sh` as well.

Reply:
SGTM